src/test/ui/linkage-attr/linkage-detect-extern-generated-name-collision.rs

```diff
@@ -4,6 +4,7 @@
 
 // build-fail
 // aux-build:def_colliding_external.rs
+// compile-flags: -Ccodegen-units=2
```

Comment:
passed in https://github.com/rust-lang/rust/pull/81942/commits/0e4d4d04636bce5b242accee74b0c95039957afd
segfaulted in rustc 3x with https://github.com/rust-lang/rust/pull/81942/commits/9159029ee97846e8c71fcc7ef339f072f6d96119
passes again with https://github.com/rust-lang/rust/pull/81942/commits/4b3adcf38f28f60e10d83464a59d869ab13460b3

This test seems to be very sensitive to unrelated factors. That might indicate some hidden bug but possibly limited to llvm9. Should I file a bug for that?